SpatialGDK/Source/SpatialGDKFunctionalTests/Private/SpatialFunctionalTestFlowController.cpp

```diff
@@ -85,28 +65,32 @@ void ASpatialFunctionalTestFlowController::CrossServerSetWorkerId_Implementation
 	WorkerDefinition.Id = NewWorkerId;
 }
 
-void ASpatialFunctionalTestFlowController::OnReadyToRegisterWithTest()
-{
-	TryRegisterFlowControllerWithOwningTest();
-}
-
 void ASpatialFunctionalTestFlowController::OnRep_OwningTest()
 {
-	TryRegisterFlowControllerWithOwningTest();
+	// Register replicated flow controllers
+	RegisterFlowController();
 }
 
-void ASpatialFunctionalTestFlowController::TryRegisterFlowControllerWithOwningTest()
+void ASpatialFunctionalTestFlowController::RegisterFlowController()
 {
-	if (!bReadyToRegisterWithTest || OwningTest == nullptr)
-	{
-		return;
-	}
-
 	OwningTest->RegisterFlowController(this);
+}
 
-	if (OwningTest->HasPreparedTest())
+void ASpatialFunctionalTestFlowController::TrySetReadyToRunTest()
+{
+	if (IsLocalController())
 	{
-		SetReadyToRunTest(true);
+		if (HasActorBegunPlay() && IsActorReady() && OwningTest->HasActorBegunPlay() && OwningTest->IsActorReady()
+			&& OwningTest->HasPreparedTest())
+		{
+			SetReadyToRunTest(true);
+		}
+		else
+		{
+			GetWorld()->GetTimerManager().SetTimerForNextTick([this]() {
+				TrySetReadyToRunTest();
```

Comment:
Do we want the limit of times we try to do this?

Reply:
I think this is caught by a generic timeout on prepareTest... hopefully

Reply:
Might we want to cap the retry count so we can be a bit more specific when this does timeout? We could log which of the conditions is not met `HasActorBegunPlay() && IsActorReady() && OwningTest->HasActorBegunPlay() && OwningTest->IsActorReady() && OwningTest->HasPreparedTest()`

Reply:
The prepare test timeout does catch this already and I've added a warning log with which conditions are not met. The list of conditions has also been reduced after the refactor that Nic suggested.